app/services/decisions/decision_tree.rb

```diff
@@ -89,32 +90,59 @@ class DecisionTree < DslDecisionTree
     # prior authority pre-draft application steps
     from(:prison_law).goto(edit: 'prior_authority/steps/authority_value')
     from(:authority_value).goto(edit: 'prior_authority/steps/ufn')
-    from(:ufn).goto(show: PRIOR_AUTHORITY_START_PAGE)
+    from(:ufn)
+      .when(-> { application.navigation_stack[-1].match?('check_answers') })
```

Comment:
I mostly concur. 

the use of the look-back on the navigation stack was just a thought as to how it can be achieved but involved moving the CYA page to the end of the stack anytime you go to that page. The look-back number needs to match the number of steps in that section flow :(, and for primary and alternative flow that falls down so, yeah, not neat at all.

I think we do need to think about this and refactor/replace it so I had already created a subtask to that effect to look at it in isolation - [subtask-1130](https://dsdmoj.atlassian.net/browse/CRM457-1130).

If you are happy to get this in to unblock - which i assume the approval indicates that? - then I, or others can pick this up as part of 1130.

Reply:
Yup, I'm happy to go ahead and merge this PR. My _mild_ preference would be to remove this part-solution for now, because we know we'll need to approach this in a different way, but above everything I'd rather we keep making progress so not too fussed either way.